tests/models/test_codegen.py

```diff
@@ -0,0 +1,105 @@
+from unittest import mock
+from unittest import TestCase
+
+from xsdata.models.codegen import Restrictions
+
+
+class RestrictionsTests(TestCase):
+    @mock.patch.object(Restrictions, "correct_required")
+    def test_post_init(self, mock_correct_required):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5205208)